### PR TITLE
feat : authentication

### PR DESCRIPTION
**Pull request by mikesch803**
1. add auth -> signup, login, guest login, logout

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,20 +9,32 @@ import {
...
 } from "./pages";
+import Mockman from "mockman-js";
+import { RequireAuth } from "./pages/auth/RequireAuth";
+import { injectStyle } from "react-toastify/dist/inject-style";
+import { ToastContainer } from "react-toastify";
 
 function App() {
+  
+  if (typeof window !== "undefined") {
+    injectStyle();
```
Comment (vrx29) [inline src/App.js:20]: Can you please tell me what is this injectStyle doing? This is something new to me

**Review by vrx29 — APPROVED** [review]
Great work, Your code is clean and easily readable. Just a suggestion try to decrease the time for toast. It is staying on screen for too long

File: src/App.js (hunk 1 above)
Comment (mikesch803) [reply src/App.js:20]: this function contains styles of toast

Comment (vrx29) [reply src/App.js:20]: cool thanks for the info

**Comment by mikesch803** [comment]
> Great work, Your code is clean and easily readable. Just a suggestion try to decrease the time for toast. It is staying on screen for too long

sure, will do the suggested changes

File: src/components/aside/Aside.css [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 .aside-searchbox {
   display: flex;
-  border: 1px solid #e0dede;
+  /* border: 1px solid #e0dede; */
```
Comment (dev-prabhat) [inline src/components/aside/Aside.css:10]: remove commented code

File: src/components/login-form/LoginForm.jsx [hunk 3]
```diff
@@ -1,50 +1,85 @@
...
     >
+      <input
+        required
+        type="text"
+        placeholder="email"
+        className="form-input mt-1"
+        name="email"
+        onChange={(e) =>
+          setFormDetail({ ...formDetail, [e.target.name]: e.target.value })
+        }
```
Comment (dev-prabhat) [inline src/components/login-form/LoginForm.jsx:41]: value attribute is missing to make it more controlled

File: src/components/login-form/LoginForm.jsx [hunk 4]
```diff
@@ -1,50 +1,85 @@
...
           required
-          type="text"
-          placeholder="email"
-          className="form-input mt-1"
-          name="email"
+          placeholder="password"
+          className="form-input flex-1"
+          name="password"
+          onChange={(e) =>
+            setFormDetail({ ...formDetail, [e.target.name]: e.target.value })
```
Comment (dev-prabhat) [inline src/components/login-form/LoginForm.jsx:50]: same value attribute is missing

**Review by dev-prabhat — APPROVED** [review]
Looks good to me, you merge and give some suggestion

**Pull request merged**